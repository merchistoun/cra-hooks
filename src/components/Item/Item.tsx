import React from "react";

interface Props {
  value: string | number;
}

export const Item = (props: Props): React.ReactElement => {
  const { value } = props;

  return <div>{value}</div>;
};
