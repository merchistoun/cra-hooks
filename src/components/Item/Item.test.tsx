import { render } from "@testing-library/react";
import { Item } from "./Item";

const mockProps = (overrides = {}) => {
  return {
    value: "TEST",
    ...overrides,
  };
};

describe("Item component", () => {
  it("should mount without error", async () => {
    const { value } = mockProps();

    const { asFragment } = render(<Item value={value} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
