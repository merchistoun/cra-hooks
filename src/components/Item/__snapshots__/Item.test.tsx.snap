// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should mount without error 1`] = `
<DocumentFragment>
  <div>
    TEST
  </div>
</DocumentFragment>
`;
